funqy docs disclaimers and usage improvements (#10693)
quarkusio/quarkus: docs/src/main/asciidoc/funqy.adoc

== Context injection

The Funqy API will usually not allow you to inject or use abstractions that

> are specific to a protocol (i.e. HTTP) or function api (i.e. AWS Lambda).  There are

Reviewer: ```suggestion
are specific to a protocol (i.e. HTTP) or function API (i.e. AWS Lambda).  There are
```

> exceptions to the rule though and you can may be able to inject

Reviewer: ```suggestion
exceptions to the rule though and you may be able to inject
```

public class GreetingFunction {

    @Funq
    public Greeting greet(Friend friend, @Context CloudEvent eventInfo) {
        System.out.println("Received greeting request from: " eventInfo.getSource());

        Greeting greeting = new Greeting();
        greeting.setMessage("Hello " + friend.getName()));
        return greeting;
    }
}
----

== Should I Use Funqy?

REST over HTTP has become a very common way to write services over the past decade.  While Funqy
has an HTTP binding it is not a replacement for REST.  Because Funqy has to work across a variety
of protocols and function cloud platforms, it is very minimalistic and constrained.  For example, if you

> use Funqy you lose the ability to link (think URIs) to the data your functions spit out.  You also

Reviewer: ```suggestion
use Funqy you lose the ability to link (think URIs) to the data your functions split out.  You also
```

Author: "spit" is correct

lose the ability to leverage cool HTTP features like `cache-control` and conditional GETs.  Many
developers will be ok with that as many won't be using these REST/HTTP features or styles. You'll
have to make the decision on what camp you are in.  Quarkus does support REST integration (through JAX-RS,

>  Spring MVC, Vert.x Web, and Servlet) with

Reviewer: Again, Spring MVC is not tested inside our integration test so better not document it untill it's tested

public class GreetingFunction {

    @Funq
    public Greeting greet(Friend friend, @Context CloudEvent eventInfo) {
        System.out.println("Received greeting request from: " eventInfo.getSource());

        Greeting greeting = new Greeting();
        greeting.setMessage("Hello " + friend.getName()));
        return greeting;
    }
}
----

== Should I Use Funqy?

REST over HTTP has become a very common way to write services over the past decade.  While Funqy
has an HTTP binding it is not a replacement for REST.  Because Funqy has to work across a variety

> of protocols and function cloud platforms, its very minimalistic and constrained.  For example, if you

Reviewer 2: ```suggestion
of protocols and function cloud platforms, it is very minimalistic and constrained.  For example, if you
```
